frontend-javascript-intermediate-functions-master/functions-and-scope.js

```diff
@@ -0,0 +1,171 @@
+
+    for (let i = 0; i < numbers.length; i++) {
+        if (numbers[i] >= 8) {
+            counting = counting + 1;
+        }
+    }
+    return counting;
+}
+
+const result1 = cumLaude(grades);
+const result2 = cumLaude([6, 4, 5]);
+const result3 = cumLaude([8, 9, 4, 6, 10]);
+
+console.log(result1);
+console.log(result2);
+console.log(result3);
+
+
+
+/* Opdracht  2: Gemiddeld cijfer */
+
+/* 2a: Script schrijven  */
+// De studenten-administratie moet ieder blok opnieuw berekenen wat het gemiddelde eindcijfer is, maar we beginnen met de grades array van hierboven.
+// Schrijf de stapjes om dit te kunnen berekenen eerst uit en vraag jezelf de volgende dingen af:
+// * Hoe wordt een gemiddelde berekend?
+// * Wat moet ik verzamelen uit de array van cijfers om uiteindelijk een gemiddelde te kunnen berekenen?
+// * Hoe zorgt ik ervoor dat ik alle waardes uit de array kan langslopen, ook als de array wel 100 entries zou bevatten?
+// Log het antwoord in de terminal.
+
+// ---- Verwachte uitkomst: 6.642857142857143
+
+
+let avNumber = 0;
+
+for (let i = 0; i < grades.length; i++) {
+    avNumber = avNumber + grades[i];
+}
+
+console.log(avNumber / grades.length);
+
+/* 2b: Omschrijven tot een herbruikbare functie */
+// Schrijf een functie genaamd averageGrade, die een array van cijfers verwacht (zoals grades) en het gemiddelde cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 2a.
+// Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
+// Log het antwoord in de terminal.
+
+// ---- Verwachte uitkomsten:
+// averageGrade(grades) geeft 6.642857142857143
+// averageGrade([6, 4, 5]) geeft xxxx
+// averageGrade([8, 9, 4, 6, 10]) geeft xxxx
+
+function averageGrade(numbers) {
+    let averageNumber = 0;
+
+    for (let i = 0; i < numbers.length; i++) {
+        averageNumber = averageNumber + numbers[i];
+    }
+
+    return averageNumber / numbers.length;
+}
+
+const result4 = averageGrade(grades);
+const result5 = averageGrade([6, 4, 5]);
+const result6 = averageGrade([8, 9, 4, 6, 10]);
+
+console.log(result4);
+console.log(result5);
+console.log(result6);
+
+
+/* 2c: Afronden op twee decimalen */
+// Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven uit de functie netjes wordt afgerond op twee decimalen.
+// Tip: Google is your best friend!
+
+
+
+
+/* Bonusopdracht: hoogste cijfer */
+
+/* 3a: Script schrijven  */
+// Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is. Je mag hier geen bestaande methoden voor gebruiken. Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
+// * Hoe kan ik iedere waarde van de array langsgaan?
+// * Op welke conditie moet ik checken?
+// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
+// Log het antwoord in de terminal.
+
+// ---- Verwachte uitkomst: 9
+
+let highestNumber = 0;
```

Comment:
In deze opdracht is het prima om als default waarde 0 te pakken, maar stel je eens voor dat hier een array binnenkomt als [-1, -4, -2, -3]. In dit voorbeeld is de hoogste waarde -1, maar met jouw functie zou er 0 uit komen terwijl die niet eens in de array voorkomt.

Nogmaals, het maakt voor deze situatie niet uit, maar het is goed om je te realiseren dat wanneer operaties als deze gaat doen op een array, dat je niet een default waarde pakt, maar altijd de eerste waarde van de array, dan weet je zeker dat je goed zit namelijk:
```suggestion
let highestNumber = grades[0];
```